sdk/metric/view/view_test.go

```diff
@@ -0,0 +1,348 @@
+				MatchInstrumentationLibrary(instrumentation.Library{
+					Name:      "bar",
+					Version:   "v1.0.0",
+					SchemaURL: "stuff.test/",
+				}),
+			},
+			match:    matchInstrument,
+			notMatch: emptyDescription,
+		},
+
+		{
+			name: "rename",
+			options: []Option{
+				MatchInstrumentName("foo"),
+				WithName("baz"),
+			},
+			match: Instrument{
+				Scope: instrumentation.Library{
+					Name:      "bar",
+					Version:   "v1.0.0",
+					SchemaURL: "stuff.test/",
+				},
+				InstrumentName:        "baz",
+				InstrumentDescription: "",
+			},
+			notMatch: emptyDescription,
+		},
+		{
+			name: "rename",
+			options: []Option{
+				MatchInstrumentName("foo"),
+				WithDescription("descriptive stuff"),
+			},
+			match: Instrument{
+				Scope: instrumentation.Library{
+					Name:      "bar",
+					Version:   "v1.0.0",
+					SchemaURL: "stuff.test/",
+				},
+				InstrumentName:        "foo",
+				InstrumentDescription: "descriptive stuff",
+			},
+			notMatch: emptyDescription,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v, err := New(tt.options...)
+			require.NoError(t, err)
+
+			t.Run("match", func(t *testing.T) {
+				got, match := v.TransformInstrument(matchInstrument)
+				assert.Equal(t, tt.match, got)
+				assert.True(t, match)
+			})
+
+			t.Run("does not match", func(t *testing.T) {
+				got, match := v.TransformInstrument(noMatchInstrument)
+				assert.Equal(t, tt.notMatch, got)
+				assert.False(t, match)
+			})
+		})
+	}
+}
+
+func TestView_matchName(t *testing.T) {
+	tests := []struct {
+		name        string
+		matchName   string
+		matches     []string
+		notMatches  []string
+		hasWildcard bool
+	}{
+		{
+			name:        "exact",
+			matchName:   "foo",
+			matches:     []string{"foo"},
+			notMatches:  []string{"foobar", "barfoo", "barfoobaz"},
+			hasWildcard: false,
+		},
+		{
+			name:        "*",
+			matchName:   "*",
+			matches:     []string{"foo", "foobar", "barfoo", "barfoobaz"},
+			notMatches:  []string{},
+			hasWildcard: true,
+		},
+		{
+			name:        "front ?",
+			matchName:   "?foo",
+			matches:     []string{"1foo", "afoo"},
+			notMatches:  []string{"foo", "foobar", "barfoo", "barfoobaz"},
+			hasWildcard: true,
+		},
+		{
+			name:        "back ?",
+			matchName:   "foo?",
+			matches:     []string{"foo1", "fooz"},
+			notMatches:  []string{"foo", "foobar", "barfoo", "barfoobaz"},
+			hasWildcard: true,
+		},
+		{
+			name:        "front *",
+			matchName:   "*foo",
+			matches:     []string{"foo", "barfoo"},
+			notMatches:  []string{"foobar", "barfoobaz"},
+			hasWildcard: true,
+		},
+		{
+			name:        "back *",
+			matchName:   "foo*",
+			matches:     []string{"foo", "foobar"},
+			notMatches:  []string{"barfoo", "barfoobaz"},
+			hasWildcard: true,
+		},
+		{
+			name:        "both *",
+			matchName:   "*foo*",
+			matches:     []string{"foo", "foobar", "barfoo", "barfoobaz"},
+			notMatches:  []string{"baz"},
+			hasWildcard: true,
+		},
```

Comment:
missing match tests:

- `**foo`
- `foo**`
- `*?foo`
- `?*foo`
- `foo*?`
- `foo?*`
- `foo*bar`
- `foo?bar`
- `meta name: .+()|[]{}^$`

Reply:
Do we need to include the meta if instrument names are restricted per [the API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule)?

> instrument-name = ALPHA 0*62 ("_" / "." / "-" / ALPHA / DIGIT)

Reply:
> Do we need to include the meta if instrument names are restricted per [the API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule)?
> 
> > instrument-name = ALPHA 0*62 ("_" / "." / "-" / ALPHA / DIGIT)

I think for a defensive programming approach we should include a test for this. It is unexpected input, but we want to ensure the system doesn't panic because of it.